apollo-backend/src/test/java/io/logz/apollo/EnvironmentTest.java

```diff
@@ -67,4 +67,97 @@ public void testGetAllEnvironments() throws ApolloClientException {
+                .findFirst();
+
+        assertThat(environmentFromApi).isPresent();
+
+        // Make sure we got the correct one
+        assertThat(environmentFromApi.get().getName()).isEqualTo(activeTestEnvironment.getName());
+        assertThat(environmentFromApi.get().getGeoRegion()).isEqualTo(activeTestEnvironment.getGeoRegion());
+        assertThat(environmentFromApi.get().getAvailability()).isEqualTo(activeTestEnvironment.getAvailability());
+        assertThat(environmentFromApi.get().getKubernetesMaster()).isEqualTo(activeTestEnvironment.getKubernetesMaster());
+        assertThat(environmentFromApi.get().getKubernetesToken()).contains("*");
+        assertThat(environmentFromApi.get().getKubernetesNamespace()).isEqualTo(activeTestEnvironment.getKubernetesNamespace());
+        assertThat(environmentFromApi.get().getIsActive()).isEqualTo(activeTestEnvironment.getIsActive());
+    }
+
+    @Test
+    public void testUpdateEnvironment() throws ApolloClientException {
+
+        ApolloTestClient apolloTestClient = Common.signupAndLogin();
+
+        // Add environment
+        Environment testEnvironment = createAndSubmitEnvironment(apolloTestClient);
+
+        // Check that environment is created
+        boolean foundTestEnvironment = apolloTestClient.getAllEnvironments().stream()
+                .anyMatch(environment -> environment.getId().equals(testEnvironment.getId()));
+        assertThat(foundTestEnvironment).isTrue();
+
+        // Create a new environment object
+        Environment newEnvironment = new Environment();
+        newEnvironment.setId(testEnvironment.getId());
+        newEnvironment.setName("eu-west-1234-logz-staging");
+        newEnvironment.setGeoRegion("eu-west-1234");
+        newEnvironment.setAvailability("staging-224");
+        newEnvironment.setKubernetesMaster(testEnvironment.getKubernetesMaster());
+        newEnvironment.setKubernetesNamespace("staging");
+        newEnvironment.setConcurrencyLimit(2);
+        newEnvironment.setServicePortCoefficient(testEnvironment.getServicePortCoefficient());
+        newEnvironment.setRequireDeploymentMessage(testEnvironment.getRequireDeploymentMessage());
+        newEnvironment.setConcurrencyLimit(testEnvironment.getConcurrencyLimit());
+        newEnvironment.setIsActive(false);
+
+        // Update previously saved environment
+        apolloTestClient.updateEnvironment(newEnvironment);
+
+        // Updated environment should be present in response
+        Optional<Environment> environmentFromApi = apolloTestClient.getAllEnvironments().stream()
+                .filter(environment -> environment.getId().equals(newEnvironment.getId()))
+                .findFirst();
+
+        assertThat(environmentFromApi).isPresent();
+
+        // Make sure we got the correct one
+        assertThat(environmentFromApi.get().getName()).isEqualTo(newEnvironment.getName());
```

Comment:
The same check of all the environment's members exist in "testGetAllEnvironments".
Maybe export the logic from there to a new method, and use there and here :)